documentation release v1 (#1012)
TimDettmers/bitsandbytes: docs/source/integrations.mdx

# Transformers

> ... TODO: to be filled out ...

Reviewer: Here we point out to relevant doc sections in transformers / peft / Trainer + explain how these are integrated 

e.g. for transformers state that you can load any model in 8-bit / 4-bit precision, for PEFT, you can use QLoRA out of the box with `LoraConfig` + 4-bit base model, for Trainer: all bnb optimizers are supported by passing the correct string in `TrainingArguments` : https://github.com/huggingface/transformers/blob/abbffc4525566a48a9733639797c812301218b83/src/transformers/training_args.py#L134 

Few references: 

- transformers: https://huggingface.co/docs/transformers/quantization#bitsandbytes
- PEFT: https://huggingface.co/docs/peft/developer_guides/quantization

# Transformers

... TODO: to be filled out ...

# PEFT

... TODO: to be filled out ...

# Trainer for the optimizers

... TODO: to be filled out ...

Here we point out to relevant doc sections in transformers / peft / Trainer + very briefly explain how these are integrated:

> e.g. for transformers state that you can load any model in 8-bit / 4-bit precision, for PEFT, you can use QLoRA out of the box with `LoraConfig` + 4-bit base model, for Trainer: all bnb optimizers are supported by passing the correct string in `TrainingArguments` : https://github.com/huggingface/transformers/blob/abbffc4525566a48a9733639797c812301218b83/src/transformers/training_args.py#L134

Reviewer: ```suggestion
e.g. for transformers state that you can load any model in 8-bit / 4-bit precision, for PEFT, you can use QLoRA out of the box with `LoraConfig` + 4-bit base model, for Trainer: all bnb optimizers are supported by passing the correct string in `TrainingArguments`'s `optim` attribute - e.g. (`paged_adamw_32bit`):
```

Few references:

> - transformers: https://huggingface.co/docs/transformers/quantization#bitsandbytes

Reviewer: ```suggestion
- [transformers documentation]( https://huggingface.co/docs/transformers/quantization#bitsandbytes)
```

> - PEFT: https://huggingface.co/docs/peft/developer_guides/quantization

Reviewer: ```suggestion
- [PEFT documentation](https://huggingface.co/docs/peft/developer_guides/quantization)
```

# Blog posts

- [Making LLMs even more accessible with bitsandbytes, 4-bit quantization and QLoRA](https://huggingface.co/blog/4bit-transformers-bitsandbytes)

> 

Reviewer: ```suggestion
- [A Gentle Introduction to 8-bit Matrix Multiplication for transformers at scale using Hugging Face Transformers, Accelerate and bitsandbytes](https://huggingface.co/blog/hf-bitsandbytes-integration)
```

> ###
> 
> For instructions how to use LLM.int8() inference layers in your own code, see the TL;DR above or for extended instruction see [this blog post](https://huggingface.co/blog/hf-bitsandbytes-integration).

Reviewer: ```suggestion
```
I would just mention the blogpost in the list of blogposts above
